catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -96,23 +144,48 @@ public NameIdentifier[] listSchemas(Namespace namespace) throws NoSuchCatalogExc
   @Override
   public Schema createSchema(NameIdentifier ident, String comment, Map<String, String> properties)
       throws NoSuchCatalogException, SchemaAlreadyExistsException {
-    throw new UnsupportedOperationException();
+    // It appears that the "default" schema suffices, so there is no need to support creating schema
+    // currently
+    throw new UnsupportedOperationException("Kafka catalog does not support schema creation");
   }
 
   @Override
   public Schema loadSchema(NameIdentifier ident) throws NoSuchSchemaException {
-    throw new UnsupportedOperationException();
+    try {
+      SchemaEntity schema = store.get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
+
+      return KafkaSchema.builder()
+          .withName(schema.name())
+          .withComment(schema.comment())
+          .withProperties(schema.properties())
+          .withAuditInfo(schema.auditInfo())
+          .build();
+
+    } catch (NoSuchEntityException exception) {
+      throw new NoSuchSchemaException(exception, "Schema %s does not exist", ident);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load schema " + ident, ioe);
+    }
   }
 
   @Override
   public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
       throws NoSuchSchemaException {
-    throw new UnsupportedOperationException();
+    if (ident.equals(defaultSchemaIdent)) {
+      throw new IllegalArgumentException("Cannot alter the default schema");
+    }
+
+    // TODO: Implement dropping schema after adding support for schema creation
+    throw new UnsupportedOperationException("Kafka catalog does not support schema alteration");
```

Comment:
Do you still need this?

Reply:
The above is for altering the default schema, which throws an `IllegalArgumentException`. 

Here, for altering other schemas, it throws an `UnsupportedOperationException`.